lib/ctblmono.gi

```diff
@@ -2176,6 +2176,74 @@ InstallMethod( IsMinimalNonmonomial,
     return false;
 end );
 
+InstallMethod( IsMinimalNonmonomial,
+    "for a non-solvable group",
+    [ IsGroup ],
+    function( K )
+
+    local info, q, p, f;
+
+    if IsSolvableGroup(K) then
+      TryNextMethod();
+    fi;
+
+    # A non-solvable group all of whose proper quotients are monomial
+    # must have every proper quotient solvable. Hence G must be simple.
+    if not IsSimpleGroup(K) then
+      return false;
+    fi;
+
+    # If in addition, every proper subgroup is monomial, then every
+    # proper subgroup must be solvable, so G must be minimal simple.
+    #
+    # A minimal non-simple group must be (by Thompson's N-Groups paper):
+    # * PSL(2,2^f) for f prime, and all such are minimal-non-M groups
+    # * PSL(2,3^f) for f an odd prime, and all such are minimal-non-M groups
+    # * PSL(3,3) which is NOT a minimal-non-M group (contains SL(2,3))
+    # * Sz(2^f) for f an odd prime, and all such are minimal-non-M groups
+    #
+    # The minimal-non-M part follows from the fact that metabelian groups
+    # are monomial groups.
+    # * For PSL(2,q) maximal subgroups are metabelian (since the non-solvable
+    #   maximals are ruled out by minimal simple)
+    # * For PSL(3,3) a subgroup isomorphic to SL(2,3) is not an M-group
+    # * For Sz(q), Suzuki 1960 shows the maximal subgroups are either the
+    #   the Sylow normalizer H, or one a few metabelian groups. The Sylow
+    #   normalizer's characters are calculated in section 11, page 126, and
+    #   are all induced from the monomial group Q, its Sylow 2-subgroup.
```

Comment:
"the the". Also, "the Sylow normalizer" makes it sound unique, but it isn't, right? And in the PR description you clarify that these are indeed normalizers of Sylow 2-groups, right? So perhaps
```suggestion
    # * For Sz(q), Suzuki 1960 shows the maximal subgroups are either the
    #   normalizer H of a Sylow 2-group Q, or one a few metabelian groups.
    #   The characters of H are calculated in section 11, page 126, and
    #   are all induced from the monomial group Q.
```